src/reading.typ

```diff
@@ -298,14 +332,25 @@
     }
     return none
   }
-  let value = read-mime(item.data.at(fmt), format: fmt, handlers: handlers)
+  let value = read-mime(item-data.at(fmt), format: fmt, handlers: handlers)
   return (
-    type: item.output_type,
     format: fmt,
-    metadata: item.metadata.at(fmt, default: none),
     value: value,
   )
 }
+/// Process rich items from the 'outputs' field in a notebook. These contain
+/// additional (meta)data besides the rich item itself.
+#let process-rich-output(item, ..args) = {
+  let processed-rich = process-rich(item.data, ..args)
+  if processed-rich == none {
+    return none
+  }
+  return (
+    type: item.output_type,
+    metadata: item.metadata.at(processed-rich.format, default: none),
+    ..processed-rich,
+  )
+}
```

Comment:
I'd prefer to avoid the split of `process-rich` into these two functions if possible

Reply:
I mean, then there is now way to reuse `process-rich` for the attachments, except by either
1. Creating a fake datastructure (dict) with `metadata` and `output_type` fields or
2. Add some checks on these fields and adjust the output depending on the input. (i.e. when its an attachment, return a dict with only `format` and `value` OR with `type: none` and `metadata: none`, when it's used on an output, the full dict like before)

I think it's cleaner to split these functions, but let me know if you want option 1 or 2.
(It's true that I dislike the extra `processed-rich == none` check too, but it's needed `-\_(¨)_/-`)

Reply:
Ah right! Let's keep the split then.